tests/python_package_test/test_sklearn.py

```diff
@@ -257,6 +257,51 @@ def test_binary_classification_with_custom_objective():
     assert ret < 0.05
 
 
+@pytest.mark.parametrize('use_weight', [True, False])
+def test_binary_classification_with_auto_early_stopping(use_weight):
+
+    X, y = load_breast_cancer(return_X_y=True)
+    n_estimators = 1000
```

Comment:
```suggestion
    n_estimators = 15
```

Please try to keep these tests as small and fast as is necessary to test the correctness of the code. We run 500+ unit tests on every build of the the Python package... the smaller they are, the more testing cycles we can get through and the faster development on the project moves.

It would also be much appreciated if you'd set `num_leaves=5` or some similar smaller-than-default value.